### ISSUE TITLE
[Version Bump Bot] Enhance the version bump bot to update the version in README and CHANGELOG

### ISSUE THREAD

**Issue by kotwanikunal** [opensearch-project/OpenSearch#8851]
**Is your feature request related to a problem? Please describe.**
- The version bump bot increments the versions once a new tag is cut/released 
- The bump does not update the versions within `README.md` and `CHANGELOG.md`

**Describe the solution you'd like**
- Support for bumps in README and CHANGELOG
- The bot should perform what is done on this PR: https://github.com/opensearch-project/OpenSearch/pull/8846

**Describe alternatives you've considered**
- Manually updating every time a new version is released.

**Additional context**

**Comment by kotwanikunal** [comment]
@samuelorji Hey Samuel. Thank you for the help.
I have assigned the issue to you. Please drop in a comment if you need any help with this.

**Comment by samuelorji** [comment]
@kotwanikunal , I'm trying to get an idea as to how I'm going to go about this.
- Add an action that updates the change log and Readme with the main branch as the base [after this step](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/version.yml#L121)


I have a question though, it seems that there are no entries in the change log and readme for 1.x versions, so are we only interested in making these changes when there is a release that starts with `2.x` ?

**Comment by kotwanikunal** [comment]
@samuelorji Yes. `1.x` is currently undergoing only security related patch changes and does not have a high velocity.
We can focus on `main` and `2.x`

**Comment by samuelorji** [comment]
@kotwanikunal, I think I have an idea on how to go about this. I have one issue however, looking at the github actions, it seems I can get the [current version](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/version.yml#L29) and [next version](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/version.yml#L32). But how do I get the previous version ?

As an example, following [this PR](https://github.com/opensearch-project/OpenSearch/pull/8846/files), the previous version was `2.8` and the current version (presumably parsed from the tag) was 2.10. So at runtime, how do I get the value of the previous version so that when I use `sed`, I use that as the token to replace, perhaps [from this file](https://github.com/opensearch-project/OpenSearch/blob/main/.ci/bwcVersions#L26)?

Or am I getting this wrong and the current version is in fact the older version and the next version is what's to be released? like in that PR, the current version was `2.8` and the next version was `2.9`?
